### ISSUE TITLE
FAR crashes ServiceHub.RoslynCodeAnalysisService32.exe

### ISSUE THREAD

**Issue by amcasey** [dotnet/roslyn#36111]
**Version Used**: eb082c8f9afb24b196d634adc3844de72af65259

**Steps to Reproduce**: Link to internal repo available upon request

**Expected Behavior**: FAR terminates

**Actual Behavior**: Gold bar indicates external process has crashed and FAR progress indicator loops forever

```
>	Microsoft.CodeAnalysis.Workspaces.dll!Roslyn.Utilities.Contract.ThrowIfNull<System.__Canon>(System.__Canon value, string message) Line 19	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.FindSymbols.SymbolFinder.VerifyForwardedTypes(System.Collections.Generic.Dictionary<Microsoft.CodeAnalysis.INamedTypeSymbol, Microsoft.CodeAnalysis.INamedTypeSymbol> equivalentTypesWithDifferingAssemblies, Microsoft.CodeAnalysis.Compilation compilation, System.Collections.Generic.HashSet<Microsoft.CodeAnalysis.INamedTypeSymbol> verifiedKeys, bool isSearchSymbolCompilation) Line 530	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.FindSymbols.SymbolFinder.VerifyForwardedTypes(System.Collections.Generic.Dictionary<Microsoft.CodeAnalysis.INamedTypeSymbol, Microsoft.CodeAnalysis.INamedTypeSymbol> equivalentTypesWithDifferingAssemblies, Microsoft.CodeAnalysis.ISymbol searchSymbol, Microsoft.CodeAnalysis.ISymbol symbolToMatch, Microsoft.CodeAnalysis.Solution solution, Microsoft.CodeAnalysis.Compilation searchSymbolCompilation, Microsoft.CodeAnalysis.Compilation symbolToMatchCompilation, System.Threading.CancellationToken cancellationToken) Line 519	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.FindSymbols.SymbolFinder.OriginalSymbolsMatchCore(Microsoft.CodeAnalysis.ISymbol searchSymbol, Microsoft.CodeAnalysis.ISymbol symbolToMatch, Microsoft.CodeAnalysis.Solution solution, Microsoft.CodeAnalysis.Compilation searchSymbolCompilation, Microsoft.CodeAnalysis.Compilation symbolToMatchCompilation, System.Threading.CancellationToken cancellationToken) Line 468	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.FindSymbols.SymbolFinder.OriginalSymbolsMatch(Microsoft.CodeAnalysis.ISymbol searchSymbol, Microsoft.CodeAnalysis.ISymbol symbolToMatch, Microsoft.CodeAnalysis.Solution solution, Microsoft.CodeAnalysis.Compilation searchSymbolCompilation, Microsoft.CodeAnalysis.Compilation symbolToMatchCompilation, System.Threading.CancellationToken cancellationToken) Line 395	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.Shared.Extensions.ITypeSymbolExtensions.FindImplementationsForInterfaceMemberAsync.AnonymousMethod__1(Microsoft.CodeAnalysis.ISymbol m) Line 178	C#
 	System.Collections.Immutable.dll!System.Linq.ImmutableArrayExtensions.FirstOrDefault<Microsoft.CodeAnalysis.ISymbol>(System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.ISymbol> immutableArray, System.Func<Microsoft.CodeAnalysis.ISymbol, bool> predicate) Line 405	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.Shared.Extensions.ITypeSymbolExtensions.FindImplementationsForInterfaceMemberAsync(Microsoft.CodeAnalysis.FindSymbols.SymbolAndProjectId<Microsoft.CodeAnalysis.ITypeSymbol> typeSymbolAndProjectId, Microsoft.CodeAnalysis.ISymbol interfaceMember, Microsoft.CodeAnalysis.Solution solution, System.Threading.CancellationToken cancellationToken) Line 213	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.FindSymbols.SymbolFinder.FindImplementationsAsync(Microsoft.CodeAnalysis.FindSymbols.SymbolAndProjectId symbolAndProjectId, Microsoft.CodeAnalysis.Solution solution, System.Collections.Immutable.IImmutableSet<Microsoft.CodeAnalysis.Project> projects, System.Threading.CancellationToken cancellationToken) Line 250	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.FindSymbols.Finders.AbstractMethodOrPropertyOrEventSymbolReferenceFinder<Microsoft.CodeAnalysis.IMethodSymbol>.DetermineCascadedSymbolsAsync(Microsoft.CodeAnalysis.FindSymbols.SymbolAndProjectId<Microsoft.CodeAnalysis.IMethodSymbol> symbolAndProjectId, Microsoft.CodeAnalysis.Solution solution, System.Collections.Immutable.IImmutableSet<Microsoft.CodeAnalysis.Project> projects, Microsoft.CodeAnalysis.FindSymbols.FindReferencesSearchOptions options, System.Threading.CancellationToken cancellationToken) Line 33	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.FindSymbols.Finders.PropertyAccessorSymbolReferenceFinder.DetermineCascadedSymbolsAsync(Microsoft.CodeAnalysis.FindSymbols.SymbolAndProjectId<Microsoft.CodeAnalysis.IMethodSymbol> symbolAndProjectId, Microsoft.CodeAnalysis.Solution solution, System.Collections.Immutable.IImmutableSet<Microsoft.CodeAnalysis.Project> projects, Microsoft.CodeAnalysis.FindSymbols.FindReferencesSearchOptions options, System.Threading.CancellationToken cancellationToken) Line 23	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.FindSymbols.Finders.AbstractReferenceFinder<Microsoft.CodeAnalysis.IMethodSymbol>.DetermineCascadedSymbolsAsync(Microsoft.CodeAnalysis.FindSymbols.SymbolAndProjectId symbolAndProjectId, Microsoft.CodeAnalysis.Solution solution, System.Collections.Immutable.IImmutableSet<Microsoft.CodeAnalysis.Project> projects, Microsoft.CodeAnalysis.FindSymbols.FindReferencesSearchOptions options, System.Threading.CancellationToken cancellationToken) Line 749	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.FindSymbols.FindReferencesSearchEngine.DetermineAllSymbolsCoreAsync.AnonymousMethod__0() Line 136	C#
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start<Microsoft.CodeAnalysis.FindSymbols.FindReferencesSearchEngine.(ref Microsoft.CodeAnalysis.FindSymbols.FindReferencesSearchEngine.<>c__DisplayClass21_1.<<DetermineAllSymbolsCoreAsync>b__0>d stateMachine) Line 316	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.FindSymbols.FindReferencesSearchEngine.DetermineAllSymbolsCoreAsync.AnonymousMethod__0()	Unknown
```

**Comment by CyrusNajmabadi** [comment]
I believe this is being fixed by the other @ryzngard here: https://github.com/dotnet/roslyn/pull/35853

Andrew#2 can you confirm? Thanks!

**Comment by ryzngard** [comment]
Yes, this looks like a dupe of #35786 and should be fixed by #35853
